things/service/src/main/java/org/eclipse/ditto/things/service/persistence/actors/enrichment/ThingEventEnricher.java

```diff
@@ -0,0 +1,385 @@
+     */
+    private Predicate<PreDefinedExtraFieldsConfig> applyPredefinedExtraFieldsCondition(
+            final Thing thing,
+            final WithDittoHeaders withDittoHeaders
+    ) {
+        return conf -> {
+            final Optional<String> optCondition = conf.getCondition();
+            if (optCondition.isEmpty()) {
+                return true;
+            }
+
+            final String rqlCondition = optCondition.get();
+
+            try {
+                final Criteria criteria = RQL_CRITERIA_CACHE.computeIfAbsent(rqlCondition, rql -> {
+                    try {
+                        return QueryFilterCriteriaFactory
+                                .modelBased(RqlPredicateParser.getInstance())
+                                .filterCriteria(rql, DittoHeaders.empty());
+                    } catch (final InvalidRqlExpressionException e) {
+                        LOGGER.warn("Invalid RQL condition <{}> - ignoring: {}", rql, e.getMessage());
+                        return null;
+                    }
+                });
+
+                if (criteria == null) {
+                    return true;
+                }
+
+                final Predicate<Thing> predicate =
+                        ThingPredicateVisitor.apply(
+                                criteria,
+                                PlaceholderFactory.newPlaceholderResolver(TIME_PLACEHOLDER, new Object()),
+                                PlaceholderFactory.newPlaceholderResolver(
+                                        HEADERS_PLACEHOLDER,
+                                        withDittoHeaders.getDittoHeaders()
+                                )
+                        );
+
+                return predicate.test(thing);
+
+            } catch (final Exception e) {
+                LOGGER.warn(
+                        "Error evaluating RQL condition <{}> for predefined extra fields - treating as match: {}",
+                        rqlCondition, e.getMessage(), e
+                );
+                return true;
+            }
+        };
+    }
+
+    private static String buildPredefinedExtraFieldsHeaderList(final JsonFieldSelector preDefinedExtraFields) {
+        return StreamSupport.stream(preDefinedExtraFields.spliterator(), false)
+                .map(JsonPointer::toString)
+                .map(JsonValue::of)
+                .collect(JsonCollectors.valuesToArray())
+                .toString();
+    }
+
+    /**
+     * Safely gets the policy enforcer stage, handling null cases defensively.
+     *
+     * @param policyId the policy ID to get the enforcer for
+     * @return a non-null CompletionStage, completed with empty Optional if the provider returns null
+     */
+    private CompletionStage<Optional<PolicyEnforcer>> getPolicyEnforcerSafely(@Nullable final PolicyId policyId) {
+        final CompletionStage<Optional<PolicyEnforcer>> stage = policyEnforcerProvider.getPolicyEnforcer(policyId);
+        if (stage == null) {
```

Comment:
This can never happen as far as I can see .. the returned CompletionStage is never null